Make tests pass with JRuby

kostyadubinin opened the pull request:
Closes #99

- [x] Fix a typo in the zip parser that causes tests to fail with JRuby.
- [x] Add support for Ruby 2.2

julik commented:
Good stuff 👌 What if we remove the safe navigation operator and replace it with something more old-fashioned - will we then become compatible with 9.0? We can then reintroduce 2.2 compatibility as well?

kostyadubinin commented:
Hey Julik, I will check if the safe navigation is the only issue.

kostyadubinin commented:
Safe navigation turned out to be the only issue with Ruby 2.2 and JRuby 9.0. Fixed in 02d84808260eedfc4abbcc0412ecad9c483d1887

Also I added the `Unreleased` section to the changelog, but if you don't use it I can remove it.

julik commented:
👍 cool. Thanks for the PR!
